server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1343,12 +1346,84 @@ public static Setting<Float> floatSetting(
         return new Setting<>(
             key,
             Float.toString(defaultValue),
-            (s) -> parseFloat(s, minValue, maxValue, key, isFiltered(properties)),
+            new FloatParser(minValue, maxValue, key, isFiltered(properties)),
             validator,
             properties
         );
     }
 
+    /**
+     * A writeable parser for float
+     *
+     */
+    public static class FloatParser implements Function<String, Float>, Writeable {
+        private float minValue;
+        private float maxValue;
+        private String key;
+        private boolean isFiltered;
+
+        public FloatParser(float minValue, float maxValue, String key, boolean isFiltered) {
+            this.minValue = minValue;
+            this.maxValue = maxValue;
+            this.key = key;
+            this.isFiltered = isFiltered;
+        }
+
+        public FloatParser(StreamInput in) throws IOException {
+            minValue = in.readFloat();
+            maxValue = in.readFloat();
+            key = in.readString();
+            isFiltered = in.readBoolean();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeFloat(minValue);
+            out.writeFloat(maxValue);
+            out.writeString(key);
+            out.writeBoolean(isFiltered);
+        }
+
+        // For testing
```

Comment:
Why this comment? Don't we need getters for this class?

Reply:
These getter is using to check these parameter is serialized and deserialized successfully or not.

Reply:
Let's the remove the comments then. I see few `For testing` comments

Reply:
ACK